src/bin/miri.rs

```diff
@@ -146,7 +147,22 @@ fn main() {
                     after_dashdash = true;
                 }
                 _ => {
-                    rustc_args.push(arg);
+                    let split: Vec<String> = arg.split("-Zmiri-seed=").map(|s| s.to_owned()).collect();
+                    if split.len() == 2 {
+                        if seed.is_some() {
+                            panic!("Cannot specify -Zmiri-seed multiple times!");
+                        }
+                        let seed_raw = hex::decode(&split[1]).unwrap();
+                        if seed_raw.len() > 8 {
+                            panic!(format!("-Zmiri-seed must be at most 8 bytes, was {}", seed_raw.len()));
+                        }
+
+                        let mut bytes = [0; 8];
+                        bytes[..seed_raw.len()].copy_from_slice(&hex::decode(&split[1]).unwrap());
+                        seed = Some(u64::from_be_bytes(bytes));
+                    } else {
```

Comment:
Could you make this a separate branch of the `match`? That makes the structure clearer.

Something like `arg if arg.starts_with("-Zmiri-seed=")` or so.